### DISCUSSION TITLE
The ability to cancel or time out a prompt

### DISCUSSION THREAD

**Discussion by rchriste** [mikaelmello/inquire#242] [Ideas]
I am wondering if there is interest in making it possible to have a prompt time out or exit after a certain amount of time or otherwise have some kind of signal to cancel the prompt.

My scenario is I am using Inquire for a time management program that I am writing, and I would love the ability to refresh every few minutes the list that I show. I use the `Select` list to show the user's "to do" items and as time passes, I would like to automatically adjust the order and put snoozed items back on the list. Right now, I am thinking that I will just have the user press Esc to refresh the list so I'm trying to get a feel for if this kind of feature or change is in line with the vision of Inquire? And if so how do you think this could or should be added?

**Comment by mikaelmello** [comment]
Hey @rchriste,

I think the concept of timeouts makes sense on a broader view, as well as an async approach to prompts too, but so far it isn't really a priority (unless there's more demand)

In your case, I think it makes more sense for your application to use something like [ratatui](https://github.com/ratatui-org/ratatui)
